arbor/include/arbor/mechanism_abi.h

```diff
@@ -0,0 +1,202 @@
+#ifndef ARB_MECH_ABI
+#define ARB_MECH_ABI
+
+#include <stddef.h>
+#include <stdint.h>
+#include <math.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// Version
+#define ARB_MECH_ABI_VERSION_MAJOR 0
+#define ARB_MECH_ABI_VERSION_MINOR 0
+#define ARB_MECH_ABI_VERSION_PATCH 1
+
+#define ARB_MECH_ABI_VERSION ((ARB_MECH_ABI_VERSION_MAJOR * 10000 * 10000) + (ARB_MECH_ABI_VERSION_MAJOR * 10000) + ARB_MECH_ABI_VERSION_PATCH)
+
+// FVM typedefs
+typedef double   arb_value_type;
+typedef float    arb_float_type;
+typedef int      arb_index_type;
+typedef uint32_t arb_size_type;
+
+typedef const char* arb_mechanism_fingerprint;
+
+// Selectors
+enum arb_mechanism_kind { point, density, reversal_potential };
+enum arb_backend_kind { cpu, gpu };
+
+// Ion state variables; view into shared_state
+typedef struct {
+    arb_value_type* current_density;
+    arb_value_type* reversal_potential;
+    arb_value_type* internal_concentration;
+    arb_value_type* external_concentration;
+    arb_value_type* ionic_charge;
+    arb_index_type* index;
+} arb_ion_state;
+
+// Event; consumed by `apply_event`
+typedef struct {
+    arb_size_type   mech_id;       // mechanism type identifier (per cell group).
+    arb_size_type   mech_index;    // instance of the mechanism
+    arb_float_type  weight;
+} arb_deliverable_event;
```

Comment:
The name doesn't quite line up with the internal struct names — this corresponds to `deliverable_event_data`, rather than `deliverable_event`. Could it be renamed to `arb_deliverable_event_data`, just for consistency?